src/api/context.js

```diff
@@ -0,0 +1,64 @@
+import * as Cookie from '../cookie';
+import * as Http from '../http';
+import Navigator from '../navigator';
+
+// consts
+import STATUS from '../status_codes';
+
+const DEFAULT_HOST = 'https://api.radar.io';
+
+class Context {
+  static getContext(callback) {
+    Navigator.getCurrentPosition((status, { latitude, longitude }) => {
+      if (status !== STATUS.SUCCESS) {
+        callback(status);
+      }
+
+      this.getContextForLocation({ latitude, longitude },
+        (status, context) => {
+          callback(status, context);
+          return;
+        }
+      )
+    });
+  }
+
+  static getContextForLocation({ latitude, longitude }, callback) {
+    const publishableKey = Cookie.getCookie(Cookie.PUBLISHABLE_KEY);
+
+    if (!publishableKey) {
+      callback(STATUS.ERROR_PUBLISHABLE_KEY);
+
+      return;
+    }
```

Comment:
at some point, I'd like to centralize all the `publishableKey` retrieving and validation in `http.js`

Reply:
Might makes sense to bite that off now